### PR TITLE
feat(apm) Add additional tracing spans

### PR DESCRIPTION
**Pull request by markstory**
Add tracing spans for query requests so we can understand a bit more of what is going on inside snuba from within tracing events.

Screen Shot 2019-11-01 at 5 05 18 PM

### CODE REVIEW

File: snuba/views.py [hunk 1]
```diff
@@ -328,7 +330,10 @@ def parse_and_run_query(dataset, request: Request, timer) -> QueryResult:
         'sample': request.query.get_sample(),
     }
 
-    return raw_query(request, query, clickhouse_ro, timer, stats)
+    with sentry_sdk.start_span(description="raw_query", op="db") as span:
+        span.set_tag("table", stats["clickhouse_table"])
```
Comment (fpacifici) [inline snuba/views.py]: please let's add the dataset class name. there may be multiple datasets loading from the same table

**Review by fpacifici — CHANGES_REQUESTED** [review]

File: snuba/views.py (hunk 1 above)
Comment (fpacifici) [reply snuba/views.py]: Also you can add the actual Clickhouse query (just like we do for postgres on sentry) by doing this 
```
span.set_tag("query", query.format_sql())
```

Or even adding the dataset name/table to the description of the span.

Comment (markstory) [reply snuba/views.py]: Sure thing. If we have the dataset & query we don't need the table, as it is redundant.

**Review by fpacifici — APPROVED** [review]

File: snuba/views.py [hunk 2]
```diff
@@ -328,7 +332,10 @@ def parse_and_run_query(dataset, request: Request, timer) -> QueryResult:
         'sample': request.query.get_sample(),
     }
 
-    return raw_query(request, query, clickhouse_ro, timer, stats)
+    with sentry_sdk.start_span(description=query.format_sql(), op="db") as span:
+        span.set_tag("dataset", dataset.__class__.__name__)
```
Comment (tkaemming) [inline snuba/views.py]:
```suggestion
        span.set_tag("dataset", type(dataset).__name__)
```

Not a big deal, but `type` is generally preferred — I forget why, might have been a bigger issue in Python 2 than 3.

File: snuba/views.py [hunk 3]
```diff
@@ -328,7 +332,10 @@ def parse_and_run_query(dataset, request: Request, timer) -> QueryResult:
...
+        span.set_tag("table", stats["clickhouse_table"])
```
Comment (tkaemming) [inline snuba/views.py]:
```suggestion
        span.set_tag("table", source)
```

**Review by tkaemming — APPROVED** [review]

**Pull request merged**